### ISSUE TITLE
Orders completed with checkoutCompleteFree method landing in "Abandoned checkouts" section

### ISSUE THREAD

**Issue by tolkachovs** [Shopify/mobile-buy-sdk-ios#703]
I'm using SDK Version 3.0.5.

After successfully completing my order using method `checkoutCompleteFree`, the order lands in "Abandoned checkouts" section on my shopify store web page. 

After further investigation I figured out that the problem is in providing billing address for the checkout. The only method to provide billing address for the checkout is `checkoutCompleteWithCreditCard` and pass billing address with `CreditCardPaymentInput` object.

In previous version of SDK (2.x) it was possible to set billing address directly to `Checkout` object, while in current version (3.x) it's not possible. 

To sum up, the question is - is this intended behavior? If not, is there any chance to make it work the "correct" way?

**Comment by dbart01** [comment]
Thanks for point this out. We'll look into providing a more intuitive interface for working with free checkout completion.

**Comment by dbart01** [comment]
Some further investigation revealed that `checkoutCompleteFree` doesn't require a `billingAddress` and is not the reason for your checkout being abandoned. Why it is being abandoned, however, is still not clear.

**Comment by tolkachovs** [comment]
Probably you've made some changes, because now it's working correctly. All my new orders, made with `checkoutCompleteFree` method are displaying on the correct tab - **Orders/All Orders**

**Comment by DanielStout5** [comment]
We are seeing this same issue - after calling checkoutCompleteFree there are no errors, but the checkouts remain in Abandoned Checkouts and never create a corresponding order.

The checkout "ready" flag never becomes true, but we can go to the webUrl and complete it manually from there.

The orders do have shipping addresses assigned.
